google-cloud-spanner/src/test/java/com/google/cloud/spanner/MockSpannerServiceImpl.java

```diff
@@ -585,6 +585,8 @@ private static void checkStreamException(
   private ConcurrentMap<String, StatementResult> partialStatementResults =
       new ConcurrentHashMap<>();
   private ConcurrentMap<String, Session> sessions = new ConcurrentHashMap<>();
+  private ConcurrentMap<String, Session> multiplexedSessions = new ConcurrentHashMap<>();
```

Comment:
This does not appear to be used. The only place where we need to check whether a session is multiplexed, is when a transaction is being executed, and that point the code just checks whether the `isMultiplexed` flag is set on that session. So I think we can safely store the multiplexed sessions in the same map as the normal sessions. That also makes methods like `listSessions` and `deleteSession` work (I think they will exclude any multiplexed session in this implementation).

Reply:
Actually a couple of tests assert the size of `sessions` list. So if we add the multiplexed session in the same list, it will increase the size by 1 and then we would have to modify all those tests to work with a different list size (if-else) logic. I felt changing it here is a cleaner solution than modifying different tests.

* google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionTest.java
* google-cloud-spanner/src/test/java/com/google/cloud/spanner/BatchCreateSessionsTest.java (`testSpannerReturnsAllAvailableSessionsAndThenNoSessions`)